### PR TITLE
Updated IDENTIFIER to allow more than 4 underscores. Also additional test code.

### PR DESCRIPTION
**Pull request by davidofferman**
Hello!

* Type: bug fix | code quality

**In raising this pull request, I confirm the following:**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/php-zephir-parser/blob/master/CONTRIBUTING.md)
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I wrote some tests for this PR.

Small description of change:
- Fixed code to allow underscore only `IDENTIFIER` to have more than 4 underscores.
- Removed `parser.c` and `scanner.c` from repo and added to `.gitignore` since they are auto generated.
- Test code for assignment operators `-=` and `*=`.
- Test code for bitwise operators.
- Test code for `constant` variable type.

Thanks

### CODE REVIEW

File: parser.mk [hunk 1]
```diff
@@ -14,7 +14,12 @@ clean: parser-clean tests-clean
 .PHONY: parser-clean
 parser-clean:
 	find . -name \*.loT -o -name \*.out | xargs rm -f
-	find ./parser -name zephir.c -o -name zephir.h | xargs rm -f
+	find ./parser \
+		   -name zephir.c  \
+		-o -name zephir.h  \
+		-o -name scanner.c \
+		-o -name parser.c  | xargs rm -f
```
Comment (sergeyklay) [inline parser.mk]: In general I would like leave scanner and parse under version control

Comment (sergeyklay) [inline parser.mk]: In general I would like leave scanner and parse under version control

Comment (davidofferman) [reply parser.mk]: Is there a reason why you want to keep them in the repo? They are always regenerated by the build scripts.

I would even suggest moving the generated files `scanner.c` and `parser.c` to a build directory so they don't get mixed up with the actual source code.

Comment (sergeyklay) [reply parser.mk]: Never mind, we can remove them.

**Comment by sergeyklay** [comment]
Thank you

**Pull request merged**